proposals/lip-define_new_transaction_schema.md

```diff
@@ -0,0 +1,258 @@
+
+
+### Type Definition
+
+| Name                | Type   | Validation                                       | Description                                                          |
+|---------------------|--------|--------------------------------------------------|----------------------------------------------------------------------|
+| `SignatureEd25519` | bytes   | Must be of length `ED25519_SIGNATURE_LENGTH`.  | Used for Ed25519 signatures.  |
+| `PrivateKeyEd25519` | bytes   | Must be of length `ED25519_PRIVATE_KEY_LENGTH`. | Used for Ed25519 private keys. |
+| `Transaction` | object | Must follow the `transactionSchema` schema with the only difference that `params` property is not serialized and contains the values of parameters of `paramsSchema` for the corresponding transaction. | An object representing a non-serialized transaction.                                        |
+
+### JSON Schema
+
+Transactions are serialized using `transactionSchema` given below.
+
+
+```java
+transactionSchema = {
+    "type": "object",
+    "required": [
+        "module",
+        "command",
+        "nonce",
+        "fee",
+        "senderPublicKey",
+        "params",
+        "signatures"
+    ],
+    "properties": {
+        "module": {
+            "dataType": "string",
+            "minLength": MIN_MODULE_NAME_LENGTH, 
+            "maxLength": MAX_MODULE_NAME_LENGTH,
+            "fieldNumber": 1
+        },
+        "command": {
+            "dataType": "string",
+            "minLength": MIN_COMMAND_NAME_LENGTH, 
+            "maxLength": MAX_COMMAND_NAME_LENGTH,
+            "fieldNumber": 2
+        },
+        "nonce": {
+            "dataType": "uint64",
+            "fieldNumber": 3
+        },
+        "fee": {
+            "dataType": "uint64",
+            "fieldNumber": 4
+        },
+        "senderPublicKey": {
+            "dataType": "bytes",
+            "length": ED25519_PUBLIC_KEY_LENGTH,
+            "fieldNumber": 5
+        },
+        "params": {
+            "dataType": "bytes",
+            "fieldNumber": 6
+        },
+        "signatures": {
+            "dataType": "array",
+            "items": {
+                "dataType": "bytes",
+                "length": ED25519_SIGNATURE_LENGTH 
+            },
+            "fieldNumber": 7
+        }
+    }
+}
+```
+
+
+#### Validation
+
+For a transaction `trs` to be valid, it must satisfy the following:
+
+* `trs` must follow the `transactionSchema`. 
+* `trs.params` is of length less than or equal to `MAX_PARAMS_SIZE` .
+
+The validity of each transaction is verified in the [static validation stage][lip-0055#block-processing] of the block processing. 
+
+
+### Serialization
+
+The serialization of an object of type `Transaction` is described in the following pseudocode.
+
+```python
+def encode(paramsSchema: LiskJSONSchema, trs: Transaction) -> bytes:
```

Comment:
As I said before, I would prefer not to pass the `paramsSchema` at all (like what you do in the signature calculation. At the end either option is fine, but I would be consistent and either always pass the params schema or always obtain it from the transaction data

Reply:
Agreed. Removed it from everywhere to be consistent. 

But now we should make sure that whenever calling this function in other LIPs we are consistent.

Reply:
I think this function should not be called by other LIPs (maybe yes, but I currently don't really see the need). But anyway, we will make sure :)